store/download_test.go

```diff
@@ -435,20 +441,136 @@ func (s *downloadSuite) TestUseDeltas(c *C) {
 	}
 
 	for _, scenario := range scenarios {
-		if scenario.exeInCore {
-			osutil.CopyFile("/bin/true", exeInCorePath, 0)
-		} else {
-			os.Remove(exeInCorePath)
-		}
+		var hostXdelta3Cmd, coreInterpCmd *testutil.MockCmd
+
+		var cleanups []func()
+
+		comment := Commentf("%#v", scenario)
+
+		// setup the env var for the scenario
 		os.Setenv("SNAPD_USE_DELTAS_EXPERIMENTAL", scenario.env)
 		release.MockOnClassic(scenario.classic)
+
+		// setup binaries for the scenario
+		if scenario.exeInCore {
+			// We need both the xdelta3 command for determining the interpreter
+			// as well as the actual interpreter for executing the basic
+			// "xdelta3 config" command.
+			// For the interpreter, since that's how we execute xdelta3, mock
+			// that as a command, but we don't need to mock the xdelta3 command
+			// in the core snap since that doesn't get executed by our fake
+			// interpreter. Mocking the interpreter and executing that as a
+			// MockCommand has the advantage that it avoids the specific ELF
+			// handling that is per-arch, etc. of the real CommandFromSystemSnap
+			// implementation.
+
+			coreInterpCmd = testutil.MockCommand(c, interpInCorePath, "")
+
+			r := store.MockSnapdtoolCommandFromSystemSnap(func(name string, args ...string) (*exec.Cmd, error) {
+				c.Assert(name, Equals, "/usr/bin/xdelta3")
+				c.Assert(args, DeepEquals, []string{"config"})
+
+				interpArgs := append([]string{"--library-path", "/some/dir/from/etc/ld.so", exeInCorePath}, args...)
```

Comment:
we can probably reduce the realism given we are mocking things anyway, just have some marker

Reply:
personally I think it is preferable to leave it in, there isn't a great reduction in complexity of the test by passing some other arguments here, and using what matches reality here is a better approximation even if things are being mocked so I feel creates a better test. I will add a comment about where these sorts of things come from though